src/components/messageCanvas/actions/copy/copy.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+
+import Icon from '../../../icon'
+import type { ThreadableMessage } from '../../../types'
+import Action from '../index'
+
+export interface CopyProps {
+  message: ThreadableMessage
+}
+
+export default function Copy(props: CopyProps) {
+  return (
+    <Action
+      message={props.message}
+      icon={<Icon name="file_copy" />}
+      onClick={(message: ThreadableMessage) => {
+        if (message.data.text) {
+          navigator.clipboard.writeText(message.data.text)
+        }
```

Comment:
I think we should indicate the text was copied using tooltip or something similar